imagescan/delta.go

```diff
@@ -321,8 +323,11 @@ func (d *deltaState) setImageScanError(i *image, err error) {
 
 	img.failures++
 	img.lastScanErr = err
-	if strings.Contains(err.Error(), "no such file or directory") {
+	if strings.Contains(err.Error(), "no such file or directory") || strings.Contains(err.Error(), "failed to get the layer") {
 		img.lastScanErr = errImageScanLayerNotFound
+		d.hostFSDisabled = true
+	} else if strings.Contains(err.Error(), "UNAUTHORIZED") || strings.Contains(err.Error(), "MANIFEST_UNKNOWN") {
```

Comment:
finding strings in error messages is huge Go anti-pattern, we should rework error logic some time in the future